contracts/minter-suite/Minters/MinterSEAV0.sol

```diff
@@ -0,0 +1,1186 @@
+     * @param _projectId The project ID being queried
+     * @return The current token ID being auctioned, or the next token ID to be
+     * auctioned if a new auction is ready to be created.
+     */
+    function getTokenToBid(uint256 _projectId) external view returns (uint256) {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Auction storage _auction = _projectConfig.activeAuction;
+        // if project has an active token auction that is not settled, return
+        // that token ID
+        if (_auction.initialized && (_auction.endTime > block.timestamp)) {
+            return _auction.tokenId;
+        }
+        // otherwise, return the next expected token ID to be auctioned
+        if (!_projectConfig.nextTokenNumberIsPopulated) {
+            revert("Next token number not populated");
+        }
+        // @dev overflow automatically checked in Solidity ^0.8.0
+        uint256 nextTokenId = (_projectId * ONE_MILLION) +
+            _projectConfig.nextTokenNumber;
+        return nextTokenId;
+    }
+
+    /**
+     * @notice View function that returns the next token ID to be auctioned
+     * by this minter for project `_projectId`.
+     * Reverts if the next token ID has not been populated for the project.
+     * @param _projectId The project ID being queried
+     * @return nextTokenId The next token ID to be auctioned by this minter
+     */
+    function getNextTokenId(
+        uint256 _projectId
+    ) external view returns (uint256 nextTokenId) {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        if (!_projectConfig.nextTokenNumberIsPopulated) {
+            revert("Next token not populated");
+        }
+        // @dev overflow automatically checked in Solidity ^0.8.0
+        nextTokenId =
+            (_projectId * ONE_MILLION) +
+            _projectConfig.nextTokenNumber;
+        return nextTokenId;
+    }
+
+    /**
+     * @dev Internal function to initialize an auction for the next token ID
+     * on project `_projectId` with a bid of `msg.value` from `msg.sender`.
+     * This function reverts in any of the following cases:
+     *   - project is not configured on this minter
+     *   - project is configured but has not yet reached its start time
+     *   - project has a current active auction that is not settled
+     *   - insufficient bid amount (msg.value < basePrice)
+     *   - no next token has been minted for the project (artist may need to
+     *     call `tryPopulateNextToken`)
+     *   - `_targetTokenId` does not match the next token ID for the project
+     * This function attempts to mint a new token and assign it to the
+     * project's next token slot. However, if the project has reached its
+     * maximum invocations on either the core contract or minter, this function
+     * will not mint a new token, and the next token slot for the project will
+     * remain empty.
```

Comment:
IIUC in this case we still revert though, no?

Reply:
I've updated these comments to help clarify in commit [fce0456](https://github.com/ArtBlocks/artblocks-contracts/pull/504/commits/fce0456f802873a234c21bccecfeaded7a163556).

This is trying to say that _after_ a new auction is created for the token in the `nextToken` slot, the function will attempt to re-fill the `nextToken` slot with a new token (by minting a new token). It will not re-fill the `nextToken` slot if the project has reached max invocations (since it needs to mint a new token to do so), but the transaction won't revert (since we still want to create an auction).

We do revert if this is called and there is no token in the `nextToken` slot at the starting state. However, the `nextToken` slot is automatically filled when an artist configures their project, adjusts max invocations, etc., so we don't expect that case to be encountered (in fact, I believe it is unreachable).

Reply:
@jakerockland - I've simplified the code that requires next token number is populated in commit [5b9eeaa](https://github.com/ArtBlocks/artblocks-contracts/pull/504/commits/5b9eeaab355a886868e9c5b1a91d3fd9874fba70).

Thanks for the quick chat today!